edxval/tests/test_api.py

```diff
@@ -687,3 +677,241 @@ def test_existing_video_in_destination_course_id(self):
+        self.assertFalse(video.encoded_videos.all().exists())
+        self.assertFalse(video.courses.all().exists())
+
+    def test_existing_video(self):
+        new_course_id = "new_course_id"
+
+        xml = self.make_import_xml(
+            video_dict={
+                "client_video_id": "new_client_video_id",
+                "duration": 0,
+            },
+            encoded_video_dicts=[
+                constants.ENCODED_VIDEO_DICT_FISH_DESKTOP,
+                {
+                    "url": "http://example.com/new_url",
+                    "file_size": 2733256,
+                    "bitrate": 1597804,
+                    "profile": "mobile",
+                },
+            ]
+        )
+        api.import_from_xml(xml, constants.VIDEO_DICT_FISH["edx_video_id"], new_course_id)
+
+        video = Video.objects.get(edx_video_id=constants.VIDEO_DICT_FISH["edx_video_id"])
+        self.assert_video_matches_dict(video, constants.VIDEO_DICT_FISH)
+        self.assert_encoded_video_matches_dict(
+            video.encoded_videos.get(profile__profile_name=constants.PROFILE_MOBILE),
+            constants.ENCODED_VIDEO_DICT_MOBILE
+        )
+        self.assertFalse(
+            video.encoded_videos.filter(profile__profile_name=constants.PROFILE_DESKTOP).exists()
+        )
+        self.assertFalse(video.courses.filter(course_id=new_course_id).exists())
+
+    def test_unknown_profile(self):
+        profile = "unknown_profile"
+        xml = self.make_import_xml(
+            video_dict=constants.VIDEO_DICT_STAR,
+            encoded_video_dicts=[
+                constants.ENCODED_VIDEO_DICT_STAR,
+                {
+                    "url": "http://example.com/dummy",
+                    "file_size": -1,  # Invalid data in an unknown profile is ignored
+                    "bitrate": 0,
+                    "profile": profile,
+                }
+            ]
+        )
+        api.import_from_xml(xml, constants.VIDEO_DICT_STAR["edx_video_id"])
+
+        video = Video.objects.get(edx_video_id=constants.VIDEO_DICT_STAR["edx_video_id"])
+        self.assertFalse(video.encoded_videos.filter(profile__profile_name=profile).exists())
+
+    def test_invalid_tag(self):
+        xml = etree.Element(
+            "invalid_tag",
+            attrib={
+                "client_video_id": "dummy",
+                "duration": "0",
+            }
+        )
+        self.assert_invalid_import(xml)
+
+    def test_invalid_video_attr(self):
+        xml = self.make_import_xml(
+            video_dict={
+                "client_video_id": "dummy",
+                "duration": -1,
+            }
+        )
+        self.assert_invalid_import(xml)
+
+    def test_invalid_encoded_video_attr(self):
+        xml = self.make_import_xml(
+            video_dict=constants.VIDEO_DICT_FISH,
+            encoded_video_dicts=[{
+                "url": "http://example.com/dummy",
+                "file_size": -1,
+                "bitrate": 0,
```

Comment:
Then should this test also be put to said other place. (I'm assuming we are testing the negative filesize)